internal/pkg/api/handleCheckin.go

```diff
@@ -1125,3 +1127,51 @@ func calcPollDuration(zlog zerolog.Logger, pollDuration, setupDuration, jitterDu
 
 	return pollDuration, jitter
 }
+
+// downloadingBytesFixed will attempt to unpack upgrade details as upgrade metadata
+// However it can encounter an error from earlier agent versions sending a string as the download rate instead of a float.
+// If it encounters this error, it will attemp to fix the download rate when unpacking the object
+// FIXME UpgradeMetadataDownloading has an UnmarshalJSON function that is not being called correctly?
```

Comment:
The downloadBytesFixed function I've added here basically reimplements: 
https://github.com/elastic/fleet-server/blob/8474293d301cc46d1821eb233c100e87426ac7c0/internal/pkg/api/metadata.go#L40-L54

But I'm really confused why our `UpgradeMetadataDownloading.UnmarshalJSON` isn't being used internally when `metadata.AsUpgradeMetadataDownloading()` is called:
https://github.com/elastic/fleet-server/blob/8474293d301cc46d1821eb233c100e87426ac7c0/internal/pkg/api/openapi.gen.go#L1691-L1695

My understanding is that the `json.Unmarshal` call should use the Unmarshaller interface